data/source/MongoDb.php

```diff
@@ -292,11 +311,19 @@ protected function _init() {
 				implode(',', $hosts)
 			);
 		}
-		parent::_init();
+
+		$server = $this->_classes['server'];
+
+		$notNull = function($value){
+			return $value !== null;
+		};
+		$uriOptions = array_filter($this->_config['uriOptions'], $notNull);
+		$driverOptions = array_filter($this->_config['driverOptions'], $notNull);
```

Comment:
The above lines dealing with filtering, can be removed, once empty options are removed.